eZ/Bundle/EzPublishCoreBundle/Converter/EzRepositoryParamConverter.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * File containing the EzRepositoryParamConverter class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ */
+
+namespace eZ\Bundle\EzPublishCoreBundle\Converter;
+
+use eZ\Publish\API\Repository\ContentService;
+use eZ\Publish\API\Repository\Exceptions\NotFoundException;
+use eZ\Publish\API\Repository\Exceptions\UnauthorizedException;
+use eZ\Publish\API\Repository\LocationService;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+use Sensio\Bundle\FrameworkExtraBundle\Request\ParamConverter\ParamConverterInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+class EzRepositoryParamConverter implements ParamConverterInterface
+{
+    private $supportedClasses = array(
+        'eZ\Publish\API\Repository\Values\Content\Location',
+        'eZ\Publish\API\Repository\Values\Content\Content'
+    );
+
+    /** @var  \eZ\Publish\API\Repository\LocationService */
+    private $locationService;
+
+    /** @var \eZ\Publish\API\Repository\ContentService */
+    private $contentService;
+
+    public function __construct( LocationService $locationService , ContentService $contentService)
+    {
+        $this->locationService = $locationService;
+        $this->contentService = $contentService;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function apply( Request $request, ParamConverter $configuration )
+    {
+        $param = $configuration->getName();
+
+        if ( !$request->attributes->has( $param ) )
+        {
+            return false;
+        }
+
+        $value = $request->attributes->get( $param );
+
+        if ( !$value && $configuration->isOptional() )
+        {
+            return false;
+        }
+
+        try
+        {
+            if ( 'eZ\Publish\API\Repository\Values\Content\Location' === $configuration->getClass() )
```

Comment:
if you prefer this style feel free, just a habit for the rest of us to get used to it :)